numpy/lib/recfunctions.py

```diff
@@ -523,6 +523,43 @@ def _drop_descr(ndtype, drop_names):
     return _fix_output(output, usemask=usemask, asrecarray=asrecarray)
 
 
+def _keep_fields(base, keep_names, usemask=True, asrecarray=False):
```

Comment:
That's true, why I made `_keep_fields` private with a leading underscore. I don't want users depending on it since it might be made obsolete in the next year or two, one way or another.